src/main/java/com/thirdcc/webapp/service/impl/UserCCInfoServiceImpl.java

```diff
@@ -91,6 +99,83 @@ public UserCCInfoDTO save(UserCCInfoDTO userCCInfoDTO) {
+            .map(userCCInfoMapper::toDto)
+            .getContent();
+        return getFullUserCCInfoList(userCCInfoDTOList, userId);
+    }
+
+    /**
+     * User CC Fish Level:
+     * First Year Session is "Junior Fish"
+     * Second Year Session is "Senior Fish"
+     * Later Year Session until Present is "Elder Fish"
+     *
+     * @param userCCInfoDTOList
+     * @param userId
+     * @return List of user CC Info Sorted by Year Session
+     */
+    private List<UserCCInfoDTO> getFullUserCCInfoList(List<UserCCInfoDTO> userCCInfoDTOList, Long userId) {
+        List<String> yearSessionWithCCFamilyRole = userCCInfoDTOList.stream()
+            .map(UserCCInfoDTO::getYearSession)
+            .collect(Collectors.toList());
+        List<UserCCInfoDTO> fullUserCCInfoDTOList = new ArrayList<>();
+        userUniInfoService.getUserUniInfoByUserId(userId)
+            .ifPresent(userUniInfoDTO -> {
+                String currentYearSession = YearSessionUtils.getCurrentYearSession();
+                String intakeYearSession = userUniInfoDTO.getYearSession();
+                String secondYearSession = YearSessionUtils.addYearSessionWithSemester(intakeYearSession, 2);
+                if (currentYearSession.compareTo(intakeYearSession) > -1) {
+                    if (!yearSessionWithCCFamilyRole.contains(intakeYearSession)) {
+                        fullUserCCInfoDTOList.add(constructCustomUserCCInfo(intakeYearSession, FishLevel.JUNIOR_FISH, userId));
+                    }
+                    else {
+                        UserCCInfoDTO userCCInfoDTO = userCCInfoDTOList.stream()
+                            .filter(userCCInfoDTO1 -> userCCInfoDTO1.getYearSession().equals(intakeYearSession))
+                            .findFirst()
+                            .get();
+                        userCCInfoDTO.setFishLevel(FishLevel.JUNIOR_FISH);
+                        fullUserCCInfoDTOList.add(userCCInfoDTO);
+                    }
+                }
+                if (currentYearSession.compareTo(secondYearSession) > -1) {
+                    if (!yearSessionWithCCFamilyRole.contains(secondYearSession)) {
+                        fullUserCCInfoDTOList.add(constructCustomUserCCInfo(secondYearSession, FishLevel.SENIOR_FISH, userId));
+                    }
+                    else {
+                        UserCCInfoDTO userCCInfoDTO = userCCInfoDTOList.stream()
+                            .filter(userCCInfoDTO1 -> userCCInfoDTO1.getYearSession().equals(secondYearSession))
+                            .findFirst()
+                            .get();
+                        userCCInfoDTO.setFishLevel(FishLevel.SENIOR_FISH);
+                        fullUserCCInfoDTOList.add(userCCInfoDTO);
+                    }
+                }
+                if (currentYearSession.compareTo(secondYearSession) > 0) {
+                    String thirdYearSession = YearSessionUtils.addYearSessionWithSemester(secondYearSession, 2) + " - Present";
```

Comment:
adding " - Present" is not clean leh.

why is this needed ya?
just return normal "2020/2021" can?

if you format the string data in this weird way, what if other modules need to consume this data?

Reply:
ok will do the change

Reply:
resolved in d5253a35de87edf18588bc0c2bae3356b925f969